pycape/function_ref.py

```diff
@@ -20,33 +20,73 @@
 
     assert fref == fref2
 """
+import enum
+from typing import Optional
+
+
+class FunctionAuthType(enum.Enum):
+    AUTH0 = 1
+    FUNCTION_TOKEN = 2
 
 
 class FunctionRef:
     """A structured reference to a Cape function."""
 
-    def __init__(self, function_id, function_hash=None):
+    def __init__(
+        self,
+        id: str,
+        hash: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> "FunctionRef":
         """Instantiate a FunctionRef.
 
         Args:
-            function_id: Required string denoting the function ID of the deployed Cape
+            id: Required string denoting the function ID of the deployed Cape
                 function. This is typically given in the output of the Cape CLI's
                 `deploy` command.
-            function_hash: Optional string denoting the function hash of the deployed
+            hash: Optional string denoting the function hash of the deployed
                 Cape function. If supplied, the Cape client will attempt to verify that
                 enclave responses include a matching function hash whenever this
                 FunctionRef is included in Cape requests.
+            token: Optional string containing a Cape function token generated
+                by the Cape CLI during `cape token`. If None, the Cape access token
+                will be used by `Cape.connect`/`Cape.run` instead.
```

Comment:
Should it be rename to `~pycape.cape.Cape.connect` and `~pycape.cape.Cape.run` since it looks like it's the new convention you are using?

Reply:
i was just experimenting here, gonna roll this out to the rest of the documentation in a separate PR

Reply:
although i did change it in the latest commits